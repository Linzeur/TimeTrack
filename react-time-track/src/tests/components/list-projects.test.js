import React from "react";
import { render, wait } from "@testing-library/react";

import { act } from "react-dom/test-utils";

import ListProjects from "../../components/list-projects";

const projects = [
  {
    category: "Education",
    client: "MINEDU",
    closed: false,
    created_at: "2019-07-11T20:14:25.250Z",
    end: null,
    estimated_cost: null,
    id: 1,
    name: "Project Rumi",
    real_cost: null,
    start: null,
    updated_at: "2019-07-11T20:14:25.250Z"
  }
];

test("testing api", async () => {
  fetch.mockResponseOnce(JSON.stringify(projects));

  const { getAllByRole } = render(<ListProjects />);
  let cards;
  await wait(() => {
    cards = getAllByRole("listitem");
  });

  expect(cards.length).toEqual(1);
});
