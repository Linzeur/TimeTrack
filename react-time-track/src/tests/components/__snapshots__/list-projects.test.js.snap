// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing api 1`] = `
<DocumentFragment>
  <main>
    <section
      class="css-10oyw64-Section"
      role="list"
    />
  </main>
</DocumentFragment>
`;
